docs/tutorials/tutorial_with_repositoryserver.rst

```diff
@@ -0,0 +1,531 @@
+``ScaleWorkload`` function scales down the ``timelog`` application before restoring the data.
+``RestoreDataUsingKopiaServer`` restores data using kopia repository server form
+s3 storage.
+
+For more information of kanister function refer
+:doc:`Kanister's parameter templating </functions>`.
+
+We are using output artifacts here to store the path of our data in s3 and
+snapshot ID that that will be used as ``backupIdentifier`` while performing restore.
+To know more about artifacts you can refer :ref:`tutorials`.
+
+Blueprint
+---------
+
+.. code-block:: yaml
+
+  $ cat <<EOF | kubectl create -f -
+  apiVersion: cr.kanister.io/v1alpha1
+  kind: Blueprint
+  metadata:
+    name: time-log-bp
+    namespace: kanister
+  actions:
+    backup:
+      outputArtifacts:
+        timeLog:
+          keyValue:
+            path: '/repo-controller/time-logger/'
+        backupIdentifier:
+          keyValue:
+            id: "{{ .Phases.backupToS3.Output.backupID }}"
+      phases:
+      - func: BackupDataUsingKopiaServer
+        name: backupToS3
+        args:
+          namespace: "{{ .Deployment.Namespace }}"
+          pod: "{{ index .Deployment.Pods 0 }}"
+          container: test-container
+          includePath: /var/log
+    restore:
+      inputArtifactNames:
+      - timeLog
+      - backupIdentifier
+      phases:
+      - func: ScaleWorkload
+        name: shutdownPod
+        args:
+          namespace: "{{ .Deployment.Namespace }}"
+          name: "{{ .Deployment.Name }}"
+          kind: Deployment
+          replicas: 0
+      - func: RestoreDataUsingKopiaServer
+        name: restoreFromS3
+        args:
+          namespace: "{{ .Deployment.Namespace }}"
+          pod: "{{ index .Deployment.Pods 0 }}"
+          image: ghcr.io/kanisterio/kanister-tools:0.92.0
+          backupIdentifier: "{{ .ArtifactsIn.backupIdentifier.KeyValue.id }}"
+          restorePath: /var/log
+      - func: ScaleWorkload
+        name: bringupPod
+        args:
+          namespace: "{{ .Deployment.Namespace }}"
+          name: "{{ .Deployment.Name }}"
+          kind: Deployment
+          replicas: 1
+  EOF
+
+Once we create a Blueprint, we can see its events by using the following command:
+
+.. code-block:: yaml
+
+  $ kubectl --namespace kanister describe Blueprint time-log-bp
+  Events:
+    Type     Reason    Age   From                 Message
+    ----     ------    ----  ----                 -------
+    Normal   Added      4m   Kanister Controller  Added blueprint time-log-bp
+
+The next CustomResource we'll deploy is an ActionSet. An ActionSet is created
+each time you want to execute any Kanister actions. The ActionSet contains all
+the runtime information the controller needs during execution. It may contain
+multiple actions, each acting on a different Kubernetes object. The ActionSet
+we're about to create in this tutorial specifies the ``time-logger`` Deployment we
+created earlier and selects the ``backup`` action inside our Blueprint.
```

Comment:
```suggestion
The next CustomResource to be deployed is an ActionSet. An ActionSet is created
whenever there is a need to execute Kanister actions. It contains all
the runtime information that is essential for the controller during execution. It can include
multiple actions, each acting on a different Kubernetes object. In this tutorial, the forthcoming ActionSet
specifies the ``time-logger`` Deployment that was previously created and selects the ``backup`` action within our Blueprint.
```